src/drawing/__tests__/DrawingAnnotations-test.tsx

```diff
@@ -232,4 +232,28 @@ describe('DrawingAnnotations', () => {
             },
         );
     });
+
+    describe('state changes', () => {
+        test.each`
+            drawn         | initialDrawn | numberOfCalls
+            ${pathGroups} | ${[]}        | ${2}
+            ${[]}         | ${[]}        | ${0}
+        `(
+            'should call update $numberOfCalls times if drawn.length is $drawn.length and initialDrawn.length is $initialDrawn.length',
+            ({ drawn, initialDrawn, numberOfCalls }) => {
+                const popupMock = { popper: { update: jest.fn() } };
+                jest.spyOn(React, 'useRef').mockImplementation(() => ({ current: popupMock }));
+                jest.spyOn(React, 'useEffect').mockImplementation(f => f());
```

Comment:
The wrapper is mounted. Do we need to mock these?

Reply:
Could you describe how I'd be able to determine if `popper.popup.update` was called if I don't mock the popup?

Reply:
In PopupReply-test.tsx, it looks like that is mounted as well but it also uses mocks in order to track whether or not `update` was called.

Reply:
Okay. Do we need to mock useEffect, as well?

Reply:
We do also need to mock `useEffect` here. When I tried removing `useEffect`, I noticed that even though the `update` calls were being logged, in the test file it indicated that the `update` call was called 0 times. I think mocking it makes it so that we're able to keep track of whether or not our mocked `update` method is called properly.